### PR TITLE
Don't issue errors for `<a/>` with target=_blank and non-external URLs

### PR DESCRIPTION
**Pull request by gfx**
`<a href="relative/path" target="_blank"/>`and `<a href="/absolute/path" target="_blank"/>` are considered safe because the link is placed in the same host, so the rule only issues for URLs with protocols.

### CODE REVIEW

File: lib/rules/jsx-no-target-blank.js [hunk 1]
```diff
@@ -33,8 +33,21 @@ module.exports = {
           var relFound = false;
           var attrs = node.parent.attributes;
           for (var idx in attrs) {
-            if (attrs[idx].name && attrs[idx].name.name === 'rel') {
-              var tags = attrs[idx].value.type === 'Literal' && attrs[idx].value.value.toLowerCase().split(' ');
+            if (!Object.prototype.hasOwnProperty.call(attrs, idx)) {
```
Comment (gfx) [inline lib/rules/jsx-no-target-blank.js]: the guard is for [guard-for-in](http://eslint.org/docs/rules/guard-for-in).

Comment (ljharb) [inline lib/rules/jsx-no-target-blank.js]: Since we have node 4+ now, let's just use `Object.keys().forEach` - then no guard is needed.

Comment (gfx) [inline lib/rules/jsx-no-target-blank.js]: in `forEach()`, only there's only`return` available to control loops, but this loop needs `return`, `continue` and `breaks`, so `forEach()` is not suitable.

Comment (ljharb) [inline lib/rules/jsx-no-target-blank.js]: In that case I'd prefer to rewrite it to use `.filter` in lieu of the continues, and `.find` in lieu of the breaks and returns.

However, to keep this PR small, we can stick with this guard (which should have been here in the first place).

File: lib/rules/jsx-no-target-blank.js [hunk 2]
```diff
@@ -33,8 +33,21 @@ module.exports = {
...
+              continue;
+            }
+            var attr = attrs[idx];
+            if (!attr.name) {
+              continue;
+            }
+            if (attr.name.name === 'href') {
+              if (attr.value.type === 'Literal' && !/^\w+:/.test(attr.value.value)) {
+                // it's safe because it is not an external link (i.e. doesn't start with a protocol)
```
Comment (ljharb) [inline lib/rules/jsx-no-target-blank.js]: `//google.com/foo` is an external link - please add a test case ensuring that [protocol-less URLs](https://www.paulirish.com/2010/the-protocol-relative-url/) still treated as external.

Comment (gfx) [inline lib/rules/jsx-no-target-blank.js]: Right!

Fixed 664e5df

File: lib/rules/jsx-no-target-blank.js (hunk 1 above)
Comment (gfx) [inline lib/rules/jsx-no-target-blank.js]: IMO the conditions are too complex so re-writing them with ES2015 features is useful.

Refactored in 89c33b6

**Pull request merged**